Rename WithTurret, WithBarrel and WithReloadingTurret to include 'Sprite' in trait name

reaperrr opened the pull request:
To make 100% clear to any modder that these are render traits (not to be confused with `Turreted`, for example) as well as for more consistent naming.

Additionally pulled out `ZOffsetFromCenter` from former `WithTurret` into a new `RenderUtils`, since there's quite a number of traits using it by now.

reaperrr committed:
Add RenderUtils

With ZOffsetFromCenter as first feature so render traits don't have to use it from WithTurret(Overlay)

reaperrr committed:
Rename WithTurret, WithBarrel and WithReloadingTurret

Make them use RenderUtils.ZOffsetFromCenter.

Mailaender commented on mods/ra/maps/allies-05a/map.yaml:
```diff
@@ -1729,7 +1729,7 @@ Rules:
 		RenderSprites:
 			Image: AGUN
-		WithTurretedSpriteBody:
+		WithSpriteTurretOverlayedSpriteBody:
```
> Error: Missing Type: WithSpriteTurretOverlayedSpriteBodyInfo

abcdefg30 commented:
Needs a rebase already.

reaperrr commented:
Rebased.

reaperrr commented:
> What is going on here?

I partially replaced the names with Notepad++ replace function instead of upgrade rules, and made a mistake when I did. I noticed later, but somehow the fix for this case ended up in the last commit instead of the 2nd.

Anyway, fixed that now to avoid more confusion.

abcdefg30 commented:
Needs a rebase.

reaperrr commented:
Rebased.

atlimit8 commented:
They would sound better if they were `With`*Part*`SpriteOverlay` and `With`*Part*`VoxelOverlay` instead of `WithSprite`*Part*`Overlay` and `WithVoxel`*Part*`Overlay`. That would fit better with `With`*Object*`SpriteBody` and `With`*Object*`VoxelBody`

reaperrr commented:
Makes sense. Will update later.

penev92 commented:
Bump.

reaperrr commented:
Updated.
I changed my mind and simplified the PR: All it does now is to insert 'Sprite' before Turret/Barrel. I think this is more consistent with `WithSpriteBody` and the voxel traits, keeps the trait names a bit shorter and the diff smaller (since I could ditch the voxel commit).

penev92 commented:
Code looks fine :+0.5:

penev92 commented:
:+1:

abcdefg30 commented:
:+1: / :white_check_mark:

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/aa452a6a596eed5e84d2145bdeeb2a16194ea713...137f3f6cf52f3c61afda0537f5002c76a6a953e9)
